app/lib/validators/nested_validator.rb

```diff
@@ -4,7 +4,7 @@
 # Check whether the passed attribute for the record has a parent.
 class NestedValidator < ActiveModel::EachValidator
   def validate_each(record, attribute, value)
-    unless value.parent.valid?
+    if value.internal_parent.nil? # We use internal parent here to allow eager loading
       record.errors[attribute] << I18n.t("errors.messages.nested")
```

Comment:
We also just check if the parent exists, rather than if it is valid. This avoids the need to hit the database to validate the uniqueness constraints on the location.